src/node_http2.cc

```diff
@@ -2813,60 +2800,76 @@ BaseObjectPtr<Http2Settings> Http2Session::PopSettings() {
   return settings;
 }
 
-Http2Settings* Http2Session::AddSettings(
-    BaseObjectPtr<Http2Settings> settings) {
-  if (outstanding_settings_.size() == max_outstanding_settings_) {
-    settings->Done(false);
-    return nullptr;
+bool Http2Session::AddSettings(Local<Function> callback) {
+  Local<Object> obj;
+  if (!env()->http2settings_constructor_template()
+          ->NewInstance(env()->context())
+              .ToLocal(&obj)) {
+    return false;
   }
-  Http2Settings* ptr = settings.get();
-  outstanding_settings_.emplace(std::move(settings));
+
+  BaseObjectPtr<Http2Settings> settings =
+      MakeDetachedBaseObject<Http2Settings>(this, obj, callback, 0);
+  if (!settings)
+    return false;
+
   IncrementCurrentSessionMemory(sizeof(*settings));
-  return ptr;
+  settings->Send();
+  outstanding_settings_.push(std::move(settings));
+  return true;
 }
 
-Http2Ping::Http2Ping(Http2Session* session, Local<Object> obj)
+Http2Ping::Http2Ping(
+    Http2Session* session,
+    Local<Object> obj,
+    Local<Function> callback)
     : AsyncWrap(session->env(), obj, AsyncWrap::PROVIDER_HTTP2PING),
       session_(session),
       startTime_(uv_hrtime()) {
+  callback_.Reset(env()->isolate(), callback);
+}
+
+Local<Function> Http2Ping::callback() const {
+  return Local<Function>::New(env()->isolate(), callback_);
```

Comment:
```suggestion
  return callback_.Get(env()->isolate());
```

(Doesn’t really make a difference, but it’s what we do throughout the codebase)